File: ChatGPT/ChatGPT/javascript/tcpServer.js
const net = require('net');

const PORT = 12345;

const server = net.createServer((socket) => {
    console.log('클라이언트가 연결되었습니다.');

    socket.on('data', (data) => {
        console.log('수신한 데이터:', data.toString());
        socket.end('서버에서 클라이언트로 응답합니다.');
    });

    socket.on('end', () => {
        console.log('클라이언트 연결이 종료되었습니다.');
    });
});

server.on('error', (err) => {
    console.error('서버 오류:', err.message);
});

server.listen(PORT, () => {
    console.log('서버가 시작되었습니다. 대기 중...');
});

File: ChatGPT/ChatGPT/go/tcpServer.go
package main

import (
	"fmt"
	"net"
)

const (
	PORT = ":12345"
)

func handleConnection(conn net.Conn) {
	defer conn.Close()

	// 클라이언트로부터 메시지 수신
	buffer := make([]byte, 1024)
	n, err := conn.Read(buffer)
	if err != nil {
		fmt.Println("수신 실패:", err)
		return
	}
	message := string(buffer[:n])
	fmt.Println("수신한 메시지:", message)
}

func main() {
	// TCP 서버 시작
	listener, err := net.Listen("tcp", PORT)
	if err != nil {
		fmt.Println("서버 시작 실패:", err)
		return
	}
	defer listener.Close()
	fmt.Println("서버 시작, 대기 중...")

	// 클라이언트 연결 대기
	for {
		conn, err := listener.Accept()
		if err != nil {
			fmt.Println("연결 수락 실패:", err)
			continue
		}
		fmt.Println("클라이언트 연결 성공")

		// 클라이언트와 통신
		go handleConnection(conn)
	}
}
